tools/scenario-tools/bin/show_hints.js

```diff
@@ -7,11 +7,19 @@ const { showHints } = require('../lib/hints.js')
 const logger = createLogger({})
 require('../lib/error-handler')
 
-const task = getCurrentTask()
-if (task === undefined) {
-  logger.error('You have not started a task.  Please run `start_task` to select your task')
-  process.exit(1)
-}
+getCurrentTask().then(task => {
```

Comment:
Switching to HTTP forced me to bite the bullet and make the IO async for progress.  I shold have done that in the first place, but the plan is to eventually rewrite these tools in golang anyway